File: sample_code/vulnerable.py
import pickle

password = "admin123"
api_key = "sk-abc123xyz"


def fetch_user(user_input: str):
    query = f"SELECT * FROM users WHERE id = {user_input}"
    return query


def dangerous_eval(expression: str):
    return eval(expression)


def unsafe_deserialize(blob: bytes):
    return pickle.loads(blob)

File: sample_code/vulnerable.js
const apiKey = "sk-live-abc123xyz";
const password = "admin123";

function fetchUser(userInput) {
  const query = `SELECT * FROM users WHERE id = ${userInput}`;
  return query;
}

function dangerousEval(expression) {
  return eval(expression);
}

function runCommand(cmd) {
  return exec(cmd);
}

module.exports = { fetchUser, dangerousEval, runCommand };
